src/main/java/org/folio/services/invoice/PoLinePaymentStatusUpdateService.java

```diff
@@ -0,0 +1,203 @@
+      logger.info("updatePoLinePaymentStatus:: The invoice was not paid or the fiscal year is unknown; " +
+        "skipping po line paymentStatus update. invoiceId={}", invoiceId);
+      return succeededFuture();
+    }
+    List<String> allPoLineIds = invoiceLines.stream()
+      .map(InvoiceLine::getPoLineId)
+      .filter(Objects::nonNull)
+      .distinct()
+      .toList();
+    if (allPoLineIds.isEmpty()) {
+      return succeededFuture();
+    }
+    logger.info("updatePoLinePaymentStatus:: Retrieving linked po lines that might need a paymentStatus update, " +
+      "invoiceId={}...", invoiceId);
+    return orderLineService.getPoLinesByIdAndQuery(allPoLineIds, this::queryToGetPoLinesWithFullyOrPartiallyPaidPaymentStatusByIds, requestContext)
+      .compose(filteredPoLines -> {
+        if (filteredPoLines.isEmpty()) {
+          logger.info("updatePoLinePaymentStatus:: No matching po line, no paymentStatus update needed, invoiceId={}",
+            invoiceId);
+          return succeededFuture();
+        }
+        logger.info("updatePoLinePaymentStatus:: There are po lines linked to the invoice that might need a paymentStatus update;" +
+          " retrieving paid invoice lines linked to the po lines...");
+        List<String> filteredPoLineIds = filteredPoLines.stream().map(PoLine::getId).toList();
+        return getInvoiceLinesByPoLineIdsAndQuery(filteredPoLineIds,
+          ids -> queryToGetRelatedPaidInvoiceLinesByPoLineIds(invoiceId, ids), requestContext)
+          .compose(relatedInvoiceLines -> {
+            if (relatedInvoiceLines.isEmpty()) {
+              logger.info("updatePoLinePaymentStatus:: No linked paid invoice line; setting paymentStatus to Awaiting Payment...");
+              return updateRelatedPoLines(relatedInvoiceLines, filteredPoLines, emptyList(), invoiceId, requestContext);
+            }
+            logger.info("updatePoLinePaymentStatus:: There are linked paid invoice lines; " +
+              "retrieving the related invoices to select only invoices with the same fiscal year as the cancelled invoice...");
+            List<String> relatedInvoiceIds = relatedInvoiceLines.stream().map(InvoiceLine::getInvoiceId).distinct().toList();
+            String invoiceQuery = "status==Paid AND fiscalYearId==" + invoiceFromStorage.getFiscalYearId();
+            return getInvoicesByIdsAndQuery(relatedInvoiceIds, invoiceQuery, requestContext)
+              .compose(relatedInvoices -> updateRelatedPoLines(relatedInvoiceLines, filteredPoLines, relatedInvoices,
+                invoiceId, requestContext));
+          });
+      });
+  }
+
+  private Future<Void> updatePoLinePaymentStatusUsingParameter(List<InvoiceLine> invoiceLines,
+      String poLinePaymentStatus, RequestContext requestContext) {
+    logger.info("updatePoLinePaymentStatusUsingParameter:: Update po line paymentStatus using parameter...");
+    if ("No Change".equals(poLinePaymentStatus)) {
```

Comment:
better to have in acq models or introduce new enum inside mod-orders code

Reply:
I tried to create it as an enum initially, within the mod-invoice raml (to avoid changing acq-models). Unfortunately the generated code did not give me the methods I needed (they are different from the enums generated from the acq-models json), and eventually I reverted to using strings. I didn't see any other case of enums used as url parameters, it is always strings. Next time I might try to use acq-models to define the enum, but I would rather not change that code at this point. I did use a pattern in the raml to restrict values.